src/esm_runscripts/filedicts.py

```diff
@@ -362,27 +407,41 @@ def _check_source_and_target(self, source_path, target_path):
             - If the ``target_path`` exists
             - If the parent dir of the ``target_path`` does not exist
         """
-
-        # Types
-        source_path_type = self._path_type(source_path)
-        target_path_type = self._path_type(target_path)
-        target_path_parent_type = self._path_type(target_path.parent)
+        # Types. Eg. file, dir, link, or None
+        try:
+            source_path_type = self._path_type(source_path)
+            target_path_type = self._path_type(target_path)
+        except TypeError as err:
+            print(err)
+            raise
```

Comment:
What is the purpose of this change?

Reply:
It was re-raising the exception. Just removed it now. Tests are still passing